peer-monitoring-service/client/src/tests/single_peer.rs

```diff
@@ -415,6 +434,72 @@ async fn test_latency_ping_failures() {
     verify_peer_latency_state(&peer_monitor_state, &validator_peer, 3, 0);
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_disconnect_from_peer() {
+    // Create the peer monitoring client and server
+    let network_id = NetworkId::Validator;
+    let (peer_monitoring_client, mut mock_monitoring_server, peer_monitor_state, time_service) =
+        MockMonitoringServer::new(vec![network_id]);
+
+    let peers = peer_monitoring_client.get_peers_and_metadata();
+
+    // Create a node config where only latency pings refresh
+    // Note: By default, this will have a max latency ping tolerated of 3
+    let node_config = config_with_latency_ping_requests();
+
+    // Spawn the peer monitoring client
+    start_peer_monitor(
+        peer_monitoring_client,
+        &peer_monitor_state,
+        &time_service,
+        &node_config,
+    )
+    .await;
+
+    // Add a connected validator peer
+    let validator_peer = mock_monitoring_server.add_new_peer(network_id, PeerRole::Validator);
+
+    // Initialize all the peer states by running the peer monitor once
+    let mock_time = time_service.into_mock();
+    let _ = initialize_and_verify_peer_states(
+        &network_id,
+        &mut mock_monitoring_server,
+        &peer_monitor_state,
+        &node_config,
+        &validator_peer,
+        &mock_time,
+    )
+    .await;
+
+    // Handle several latency ping requests with bad responses
+    for i in 0..node_config
+        .peer_monitoring_service
+        .latency_monitoring
+        .max_latency_ping_failures
+    {
+        elapse_latency_update_interval(node_config.clone(), mock_time.clone()).await;
```

Comment:
nit: add the comment for consistency? e.g., `// Elapse enough time for a latency ping update`? 😄